Quadriga/src/main/java/edu/asu/spring/quadriga/service/workbench/StatisticsSettingsService.java

```diff
@@ -0,0 +1,118 @@
+package edu.asu.spring.quadriga.service.workbench;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.asu.spring.quadriga.dao.workbench.settings.IStatisticDAO;
+import edu.asu.spring.quadriga.dto.ProjectStatisticsDTO;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.profile.validator.NotEmptyList;
+
+/**
+ * this class holds the list of statistics settings for a project and used in
+ * the controller as a modelattribute
+ * 
+ * @author Ajay Modi
+ *
+ */
+@Service
+public class StatisticsSettingsService {
+
+    @Autowired
+    private IStatisticDAO statisticDao;
+
+    @Transactional
+    public void updateRow(String projectId, String name, Boolean isChecked)
+            throws QuadrigaStorageException {
+        statisticDao.upsertStatisticsSettings(projectId, name, isChecked);
+    }
+
+    @Transactional
+    public void updateSettings(String projectId, String[] names)
+            throws QuadrigaStorageException {
+        List<String> statisticsSettingsList = getStatisticsSettingsList(projectId);
+        List<String> namesList = Arrays.asList(names);
+        if (statisticsSettingsList != null && statisticsSettingsList.size() > 0) {
+            for (String name : statisticsSettingsList) {
+                if (namesList.contains(name)) {
+                    updateRow(projectId, name, true);
+                } else {
+                    updateRow(projectId, name, false);
+                }
+            }
+        }
+    }
+
+    @Transactional
+    public List<ProjectStatisticsDTO> getStatisticsSettingsDTOList(

```

Comment:
Your manager should not pass on DTOs. DTOs or between manager and dao class. You would either create a domain object (and its interface) that this method passes on and map your DTO to the domain object, or, if your DTO and domain object look the same, you could just create the interface for your domain object and have the DTO implement that. Then you can pass on the DTO, but controllers won't know, because your return type is the interface. This way, if our domain objects ever need to have more functionality than the DTOs (e.g. a method that calculates the best text color based on the chart color), then we can simply add the domain class/mapping part to the manager without having to change the structure of our classes. Does this make sense?